homeassistant/components/seventeentrack/sensor.py

```diff
@@ -111,242 +95,150 @@ async def async_setup_entry(
 ) -> None:
     """Set up a 17Track sensor entry."""
 
-    client = hass.data[DOMAIN][config_entry.entry_id]
+    coordinator: SeventeenTrackCoordinator = hass.data[DOMAIN][config_entry.entry_id]
+
+    @callback
+    def _async_create_remove_entities():
+        remove_packages(hass, coordinator.account_id, coordinator.data.old_packages)
+
+        async_add_entities(
+            SeventeenTrackPackageSensor(
+                coordinator, t_number, p_data.friendly_name, p_data.status
+            )
+            for t_number, p_data in coordinator.data.new_packages.items()
+            if not (not coordinator.show_delivered and p_data.status == "Delivered")
+        )
+
+        for tracking_number, package_data in coordinator.data.new_packages.items():
+            if (
+                package_data.status == VALUE_DELIVERED
+                and not coordinator.show_delivered
+            ):
+                notify_delivered(
+                    hass,
+                    package_data.friendly_name,
+                    tracking_number,
+                )
+
+    async_add_entities(
+        SeventeenTrackSummarySensor(status, coordinator)
+        for status in coordinator.data.summary
+    )
+
+    _async_create_remove_entities()
 
-    data = SeventeenTrackData(
-        client,
-        async_add_entities,
-        DEFAULT_SCAN_INTERVAL,
-        config_entry.options[CONF_SHOW_ARCHIVED],
-        config_entry.options[CONF_SHOW_DELIVERED],
-        str(hass.config.time_zone),
+    config_entry.async_on_unload(
+        coordinator.async_add_listener(_async_create_remove_entities)
     )
-    await data.async_update()
 
 
-class SeventeenTrackSummarySensor(SensorEntity):
+class SeventeenTrackSummarySensor(
+    CoordinatorEntity[SeventeenTrackCoordinator], SensorEntity
+):
     """Define a summary sensor."""
 
     _attr_attribution = ATTRIBUTION
     _attr_icon = "mdi:package"
     _attr_native_unit_of_measurement = "packages"
 
-    def __init__(self, data, status, initial_state) -> None:
-        """Initialize."""
-        self._attr_extra_state_attributes = {}
-        self._data = data
-        self._state = initial_state
+    def __init__(
+        self,
+        status: str,
+        coordinator: SeventeenTrackCoordinator,
+    ) -> None:
+        """Initialize the sensor."""
+        super().__init__(coordinator)
         self._status = status
-        self._attr_name = f"Seventeentrack Packages {status}"
-        self._attr_unique_id = f"summary_{data.account_id}_{slugify(status)}"
+        self._attr_name = f"Seventeentrack Packages {self._status}"
+        self._attr_unique_id = f"summary_{coordinator.account_id}_{self._status}"
 
     @property
     def available(self) -> bool:
         """Return whether the entity is available."""
-        return self._state is not None
+        return self.coordinator.data.summary[self._status]["quantity"] is not None
```

Comment:
When this is unavalable, is summary[status] even set? Shouldn't it be

```suggestion
        return self._status in self.coordinator.data.summary
```